zookeeper-server/src/main/java/org/apache/zookeeper/MultiOperationRecord.java

```diff
@@ -126,7 +127,10 @@ public void deserialize(InputArchive archive, String tag) throws IOException {
                 case ZooDefs.OpCode.createContainer:
                     CreateRequest cr = new CreateRequest();
                     cr.deserialize(archive, tag);
-                    add(Op.create(cr.getPath(), cr.getData(), cr.getAcl(), cr.getFlags()));
+                    EnumSet<Op.CreateFlags> createFlags = h.getType() == ZooDefs.OpCode.create2
+                        ? EnumSet.of(Op.CreateFlags.WANT_STAT)
```

Comment:
Can we have a constant even for this case? We are going to reduce memory allocations

Reply:
Done.

Adding that odd case to the visible API felt a bit awkward, so I have attached it to the `MultiOperationRecord` implementation.  (Of course, the "odd case"-ness is debatable since we only have a single flag :)  Let me know if you'd prefer to have it in the enum.  (In which case I would also appreciate a suggestion for a non-clashing name!)

I have also wrapped both `EnumSet`s with `Collections.unmodifiableSet`, as the former are in fact mutable.